internal/charmstore/server.go

```diff
@@ -103,6 +103,30 @@ type ServerParams struct {
 	// RunBlobStoreGC holds whether the server will run
 	// the blobstore garbage collector worker.
 	RunBlobStoreGC bool
+
+	// BlobStore holds the type of blobstore to use.
+	BlobStore string `yaml:"blobstore"`
+
+	// SwiftAuthURL holds the swift auth url, equivalent to OS_AUTH_URL.
+	SwiftAuthURL string `yaml:"swift-auth-url"`
```

Comment:
I think it would be better if a blobstore implementation was passed in here rather than all the possible configurations for one, That does however require some code re-arrangement to work I would think.